programs/ssl/ssl_client2.c

```diff
@@ -358,6 +362,8 @@ struct options
     int etm;                    /* negotiate encrypt then mac?              */
 } opt;
 
+int query_config( const char *config );
```

Comment:
I assume that this line has been put here and in the other program in order not to introduce a header file, which may mess with the concept of the public API. Nevertheless, this looks a bit odd, and I would opt for a cleaner solution. I think that @hanno-arm's suggestion to put the code in a separate program would solve the issue in a simpler way. I would also think, how this change impacts the value of the program as an example - maybe we should avoid addind non-critical complexity to the example programs.

Reply:
This is here because I did not think it was necessary to add a header file because of a 1 line function declaration. I am not particularly worried about the API because this is an example application, not the library.

>I think that @hanno-arm's suggestion to put the code in a separate program would solve the issue in a simpler way.

Please see the discussion with @sbutcher-arm and @hanno-arm regarding stand-alone app vs ssl_client2/ssl_server2 extension.